rfcs/0166-Sign-public-S3-urls.md

```diff
@@ -0,0 +1,49 @@
+# RFC 166 - Sign Public S3 URLs
+* Comments: [#166](https://github.com/taskcluster/taskcluster-rfcs/pull/166)
+* Proposed by: @ricky26
+
+# Summary
+
+At the moment, there are two S3 artefact buckets: one for public artefacts and
+one for private artefacts. We will introduce a deploy-time parameter which
+configures whether URL signing is used for the public artefacts bucket.
+
+## Motivation
+
+This is primarily targeted at private installations of TaskCluster. As part of 
+running a private TaskCluster installation, it would be ideal to have no
+publicly accessible S3 buckets.
+
+At the moment, the public bucket needs to be exposed globally as we redirect to
+public S3 URLs. For the private bucket, we avoid this requirement by using S3
+signed URLs.
+
+We already have two code paths for artefact URLs and this would just mean always
+using signed URLs in private TaskCluster installations.
+
+# Details
+Signed S3 URLs have a drawback compared to public URLs: they have an expiry
+time. Introducing this change will lead to a situation where artefact URLs
+fetched for public artefacts from a TaskCluster installation with this flag will
+expire whereas otherwise they would not.
```

Comment:
I think this is OK -- part of the reason to use redirects and not return a URL is to discourage storing the resulting URL, so hopefully nobody's doing that.  For example, [Firefox-CI test tasks](https://firefox-ci-tc.services.mozilla.com/tasks/dmCrgYXVTdK82mRkTat6XQ) are given the URL from which to download the browser installer as a URL for the `queue.getArtifact` API method.